File: modelsMongo/Movie.js
import mongoose, {model, Schema} from "mongoose";

const MovieSchema = new Schema({
    tmdbId: {
        type: String,
        required: true
    },
    reviewsId: {
        type: [mongoose.Types.ObjectId],
        ref: 'Review',
        default: []
    },
    title: {
        type: String,
        required: true
    },
    genres: {
        type: [{'id': Number, 'name': String}],
        required: true
    },
    releaseDate: {
        type: Date,
    },
    overview: {
        type: String,
    },
    imgUrl: {
        type: String,
    },
    popularity: {
        type: Number,
        required: true
    },
    youTubeKey: {
        type: String,
    },

    ratingCount: {
        type: Number,
        default: 0
    },
    averageRating: {
        type: Number,
        default: 0
    },
}, {
    timestamps: true,
});

export default model('Movie', MovieSchema);

File: tfAPI/tfService/app/schemas/movie.py
from pydantic import BaseModel
from typing import List


class Genre(BaseModel):
    id: int
    name: str
    _id: str


class Movie(BaseModel):
    _id: str
    tmdbId: str
    reviewsId: List[str]
    title: str
    genres: List[Genre]
    releaseDate: str
    overview: str
    imgUrl: str
    popularity: float
    youTubeKey: str
    ratingCount: List[int]
    averageRating: float
    createdAt: str
    updatedAt: str